File: binary_search/leetcode704_binarysearch/leetcode704_iteration.py
# 704


def search(nums, target: int) -> int:
    left = 0
    right = len(nums) - 1

    while left <= right:
        mid = left + (right - left) // 2

        if nums[mid] < target:
            left = mid + 1
        elif target < nums[mid]:
            right = mid - 1
        else:
            return mid

    return -1

File: binary_search/leetcode704_binarysearch/leetcode704_iteration.js
// nums = [-1,0,3,5,9,12], target = 9

const search = function (nums, target) {
	let left = 0;
	let right = nums.length - 1;
	while (left <= right) {
		// let mid = Math.floor((left + right) / 2);
		let mid = left + Math.floor((right - left) / 2);
		if (nums[mid] > target) {
			right = mid - 1;
		} else if (nums[mid] < target) {
			left = mid + 1;
		} else {
			return mid;
		}
	}
	return -1;
};

console.log(search([-1, 0, 3, 5, 9, 12], 9));
